open-authoring/OpenAuthoringErrorInterpreter.ts

```diff
@@ -0,0 +1,122 @@
+import OpenAuthoringError from "./OpenAuthoringError";
+import OpenAuthoringContextualError from "./OpenAuthoringContextualError";
+import { Actions } from "./OpenAuthoringModalContainer";
+import { isForkValid } from "./github/api";
+
+export default async function interpretError(error: OpenAuthoringError) : Promise<OpenAuthoringContextualError> {
+    if (!error || !error.code) {
+        return new OpenAuthoringContextualError(
+            false, // should it be presented as a modal? (if not present a toast)
+            "Error", // title
+            "Failed to interpret error. No context provided.", // message (the only thing a toast will present)
+            [{ 
+                message: "Continue",
+                action: Actions.authFlow
+            },
+            { 
+                message: "Cancel",
+                action: Actions.refresh
+            }], // Action buttons
+            false // clear preview cookies
+        )
+    }
+    
+    switch (parseInt(error.code.toString()[0])) {
+        case 4: {
+            return await interpretClientError(error)
+        }
+        case 5: {
+            return interpretServerError(error)
+        }
+    }
+
+    return new OpenAuthoringContextualError(
+        false,
+        "Error",
+        "Failed to interpret error.",
```

Comment:
Also feels a bit developer'ey for a user error message

Reply:
those messages are more dev'ey to let us know that we have an error that isn't being interpreted properly, but we can certainly make it more generic / understandable to be user friendly

Reply:
This is shown to the user in the modal though right? Ideally I'd say we'd want to do a console.warn for the developer message and then something more generic in the modal